### PR TITLE
Allow to change default config files extension

### PR DESCRIPTION
**Pull request by jay7x**

### CODE REVIEW

File: manifests/configfile.pp [hunk 1]
```diff
@@ -40,6 +43,7 @@
   Optional[String] $content = undef,
   Stdlib::Absolutepath $config_dir = $caddy::config_dir,
   Optional[Stdlib::Absolutepath] $enable_dir = $caddy::config_enable_dir,
+  String $file_extension = $caddy::config_file_extension,
```
Comment (bastelfreak) [inline manifests/configfile.pp:46]: do we want to allow empty string here?

Comment (jay7x) [reply manifests/configfile.pp:46]: yes, I did it intentionally.. if someone would like to include any file in a config directory irregardless of extension

File: manifests/configfile.pp [hunk 2]
```diff
@@ -52,7 +56,9 @@
     default  => 'file',
   }
 
-  file { "${config_dir}/${title}.conf":
+  $filename = "${title}${file_extension}"
```
Comment (bastelfreak) [inline manifests/configfile.pp:59]: should this be `"${title}.${file_extension}"`, or should we maybe enforce that `$file_extension` starts with a dot?

Comment (jay7x) [reply manifests/configfile.pp:59]: This is to allow to include any file irregardless of the extension

Comment (jay7x) [reply manifests/configfile.pp:59]: Though, I see the point.. maybe it should be `Variant[Enum[''], Pattern[/^\..*$/]]` to allow both empty string and anything starting with a dot. I hope nobody would want to use underscore or dash here 🤞🏻.

Comment (jay7x) [reply manifests/configfile.pp:59]: Then, maybe it's better to rename it from "extension" to "suffix", as it reflects the behavior better.

**Review by bastelfreak — APPROVED** [review]

**Pull request merged**